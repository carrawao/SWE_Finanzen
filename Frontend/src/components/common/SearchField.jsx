import React from 'react';
import TextField from '@mui/material/TextField';
import {InputAdornment, IconButton} from '@mui/material';
import SearchIcon from '@mui/icons-material/Search';
import PropTypes from "prop-types";

/**
 * Component for the search field in the screens' header
 * @returns {JSX.Element}
 * @constructor
 */
const SearchField = props => {
  /**
   * Get result of shares and cryptos in the search field
   * @param query
   * @returns {Promise<Response<any, Record<string, any>, number>>}
   */
  const search = async query => {
    props.setSearchQuery(query);
    if (query !== '') {
      try {
        return await fetch(`http://localhost:3001/search?text=${query}`, {mode:'cors'})
          .then(response => response.json())
          .then(searchResult => {
            let firstResults = searchResult.slice(0, 10);
            props.setSearchResult(firstResults);
          });
      }
      catch (e) {
        console.log('fetching failed === ', e);
      }
    } else {
      props.setSearchResult([]);
    }
  }

  return (
    <TextField
      id='standard-basic'
      variant='standard'
      placeholder='Search Asset'
      className='d-flex flex-grow-1'
      sx={{
        '& .MuiInputBase-input': {
          '&::-webkit-input-placeholder': {
            color: 'black'
          },
        },
        '& .MuiInput-underline': {
          '&:after': {
            borderBottomColor: '#493f35',
          },
          '&:focus::after': {
            borderBottomColor: '#493f35',
          },
        },
        '& .MuiInputBase-root:before': {
          borderBottom: '1px solid #493f35',
        },
      }}
      InputProps={{
        startAdornment: (
          <InputAdornment position='start'>
            <IconButton className='pe-0'>
              <SearchIcon style={{color: '#493f35'}}/>
            </IconButton>
          </InputAdornment>
        ),
      }}
      value={props.searchQuery}
      onChange={event => search(event.target.value)}
    />
  );
}

SearchField.propTypes = {
  searchQuery: PropTypes.string,
  setSearchQuery: PropTypes.func,
  searchResult: PropTypes.array,
  setSearchResult: PropTypes.func,
};

export default SearchField;