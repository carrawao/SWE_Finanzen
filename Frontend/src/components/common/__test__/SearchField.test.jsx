import renderer from 'react-test-renderer';
import SearchField from '../SearchField';

it('changes the class when hovered', () => {
  const component = renderer.create(
    <SearchField/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  renderer.act(() => {
    tree.props.onMouseEnter();
  });
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  renderer.act(() => {
    tree.props.onMouseLeave();
  });
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});