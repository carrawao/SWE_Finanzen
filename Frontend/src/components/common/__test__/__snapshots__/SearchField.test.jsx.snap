// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the class when hovered 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1tg2a5i-MuiFormControl-root-MuiTextField-root"
>
  <div
    className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart css-ghsjzk-MuiInputBase-root-MuiInput-root"
    onClick={[Function]}
  >
    <div
      className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
    >
      <span
        className="notranslate"
      >
        â€‹
      </span>
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium pe-0 css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="SearchIcon"
          focusable="false"
          style={
            Object {
              "color": "#493f35",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
      </button>
    </div>
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1x51dt5-MuiInputBase-input-MuiInput-input"
      disabled={false}
      id="standard-basic"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Search Asset"
      required={false}
      type="text"
    />
  </div>
</div>
`;
